src/include/btree_inline.h

```diff
@@ -2269,6 +2269,31 @@ __wt_btcur_bounds_early_exit(
     return (0);
 }
 
+/*
+ * __wt_cbt_clear_all_deleted_items_flag --
+ *     Reset the page all deleted items flag if it has any data other than deleted.
+ */
+static inline void
+__wt_cbt_clear_all_deleted_items_flag(
+  WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *first_upd)
+{
+    if (cbt->upd_value->tw.stop_txn != WT_TXN_MAX) {
+        if (__wt_txn_upd_value_visible_all(session, cbt->upd_value))
+            ++cbt->page_obsolete_deleted_count;
+        /*
+         * If the selected tombstone is not same as the most recent update in the update list
+         * indicates that there are newer updates in the update list that are either not committed
+         * or not visible to the current transaction. In this scenario, consider the page is having
+         * non deleted updates.
+         */
+        else if (F_ISSET(cbt, WT_CBT_ALL_DELETED_ITEMS) && first_upd &&
```

Comment:
first_upd != NULL?

Reply:
Why this check is necessary? We should do update restore eviction on this page and it should still work, right?

Reply:
Changed the condition. Regarding update restore eviction, we are in the path of scan and if we found a non visible update on the page, we do not perform the eviction. Even if we perform the eviction, the follow of update restore eviction is not changed and it works.

Reply:
This makes sense. Even we do an update restore eviction, we cannot move this page to memory so we don't get any benefit of search improvement.